pkg/server/init.go

```diff
@@ -550,7 +537,7 @@ func (s *initServer) attemptJoinTo(ctx context.Context, addr string) (*initState
 		StoreID:   storeID,
 	}
 
-	firstEngine := s.mu.inspectState.uninitializedEngines[0]
+	firstEngine := s.inspectedDiskState.uninitializedEngines[0]
 	if err := kvserver.InitEngine(ctx, firstEngine, sIdent); err != nil {
 		return nil, err
```

Comment:
Shouldn't this also be a fatal? It seems like we could rearrange a bit to make this error handling cleaner. We'd want only one place that collects an error and fatals on it, rather than peppering this method (which has an error in its return signature) with fatals.

- `attemptJoinTo` returns the JoinResponse and an error
- if not error, `startJoinLoop` passes the JoinResponse to `initStoreFromJoin(...)` (name tbd) and all errors returned from that are fatals